enhancements/microshift/etcd-as-a-transient-systemd-unit.md

```diff
@@ -0,0 +1,202 @@
+While this is not ideal, decoupling etcd into its own process to resolve the dependency conflict is the first step and getting the etcd Go build version aligned back to 1.16 can be a future goal.
+
+##### RPM Pros
+* This scenario should incur the least amount of non-etcd-related overhead since the binary is being run directly.
+* Integration is straight-forward; the etcd rpm is another MicroShift dependency, embedded into the rpm-ostree and distributed the same way as other dependencies.
+* Clear, consistent architecture: There is only either RPM-installed content or content hosted on the cluster - no third way of installing/running things.
+    
+##### RPM Cons
+* New plumbing will need to be created to (or potentially existing will need to be updated/refurbished to) build etcd and package it into an RPM.
+* etcd will need to be built with a newer version of Golang (1.19) than is currently expected from upstream (1.16); etcd will be built under the MicroShift build root, locking the Golang version.
+
+### Workflow Description
+
+TODO
+
+#### Variation [optional]
+
+TODO
+
+### API Extensions
+
+TODO
+
+### Implementation Details/Notes/Constraints [optional]
+
+TODO
+
+### Risks and Mitigations
+
+TODO
+
+### Drawbacks
+
+TODO
+
+## Design Details
+
+### Open Questions
+
+#### RPM Build Cadence
+How do we get to pre-release version built out for every sprint?
+* How do we get fast-fix changes to customers that need/want them before a proper release?
+
+#### Naming
+Do we want to have it be just `etcd` or `microshift-etcd` or other?
+* Do we want to somehow indicate that this is specifically for MicroShift and not for general use; does that matter?
+
+#### Versioning
+
+* How should we version the etcd RPM; should we follow the OpenShift versioning for etcd.
+  * New versioning scheme that maps to an OpenShift/Upstream version
+  * Go with the etcd version: v3.5.4
+    * How do we reflect a patch or cherrypick update from OpenShift/etcd?
+  * OpenShift version: v4.12
+    * What is the release cadence for the etcd RPM/MicroShift as a whole? v4.12.1, v4.12.2, etc.
+    * How do we reflect an update to the etcd version for a particular OpenShift release branch e.g 4.12 gets updated from etcd 3.5.4 to 3.5.5?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+TODO
+
+### Graduation Criteria
+
+TODO
+
+#### Dev Preview -> Tech Preview
+
+TODO
+
+#### Tech Preview -> GA
+
+TODO
+
+#### Removing a deprecated feature
+
+TODO
+
+### Upgrade / Downgrade Strategy
+
+TODO
+
+### Version Skew Strategy
+
+TODO
+
+### Operational Aspects of API Extensions
+
+TODO
+
+#### Failure Modes
+
+TODO
+
+#### Support Procedures
+
+TODO
+
+## Implementation History
+
+TODO
+
+## Alternatives
```

Comment:
Another option would be to compile etcd into a [go plugin](https://pkg.go.dev/plugin) and then continue to execute it in a goroutine. This would allow you to have a completely separate build chain for the binary, while not changing how etcd is run today. If necessary you can embed the plugin into the main Microshift binary. The installer is taking this approach with Terraform, since the various Terraform providers for different clouds often have conflicting dependencies.

Reply:
Doug pointed out that you can still only have one version of each dependency otherwise the symbols conflict. (Additionally, the plugin has to be built with the exact same runtime version.) This presumably works in the installer because only one plugin is loaded at a time, and the dependency conflicts are between different Terraform plugins not between the installer and the plugin. So this won't work, but perhaps it's worth documenting why in the alternatives section.